### PR TITLE
Spec for Package Source Mapping Package Management Actions in the PM UI

### PR DESCRIPTION
**Pull request by donnie-msft**
Package Source Mapping Package Management Actions in the PM UI ([shortcut to rendered spec](https://github.com/NuGet/Home/blob/dev-donnie-msft-specSourceMappingActionSpec/proposed/2023/package-source-mapping-actions-PMUI.md))

Spec for: https://github.com/NuGet/Home/issues/11366
Implementation PR: https://github.com/NuGet/NuGet.Client/pull/5355

### CODE REVIEW

**Review by jeffkl — APPROVED** [review]
Great spec!

File: proposed/2023/package-source-mapping-actions-PMUI.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+The customer already has the option to disable this preview dialog, so if they choose to disable it, there will be nothing telling them what new mappings were created.
+
+### Technical Explanation
+
+Package Management action logic will pass down a new mapping package ID and source name to Restore. Preview Restore Utilities will look for this new mapping and use it in the preview restore: By appending 2 package patterns, a "`*`" and the new package ID, along with the new mapping source name to Settings in memory.
+
+If the customer has enabled the Preview Window, then they must consent to the preview result. It's not required to enable the Preview Window in order for mappings to be automatically created, but if it's shown and cancelled, then no mappings will be written.
+
+Once preview restore determines the added packages, those package IDs will be checked against existing mappings to calculate deltas. The deltas will be written to disk in the applicable `NuGet.Config` after a successful preview restore, and immediately before executing Restore.
```
Comment (nkolev92) [inline proposed/2023/package-source-mapping-actions-PMUI.md]: During the in person review, I mentioned that it'd be great to have examples about the deltas and how that works. 
Ex. When would the mappings happen vs when they wouldn't etc. 
A truth table-ish thing would be great.

Comment (donnie-msft) [reply proposed/2023/package-source-mapping-actions-PMUI.md]: Take a look at the matrix I added...

File: proposed/2023/package-source-mapping-actions-PMUI.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+If the customer has enabled the Preview Window, then they must consent to the preview result. It's not required to enable the Preview Window in order for mappings to be automatically created, but if it's shown and cancelled, then no mappings will be written.
+
+Once preview restore determines the added packages, those package IDs will be checked against existing mappings to calculate deltas. The deltas will be written to disk in the applicable `NuGet.Config` after a successful preview restore, and immediately before executing Restore. 
+
+The Global Packages Folder (GPF) will not be checked for whether these packages already exist nor from which package source they were retrieved.
+
+#### Table: How package source configuration affects automatic Package Source Mappings
+
+| Dependency Type | Existing Mapping | Selected Source | Package on Source(s) | New Mapping Source |
```
Comment (nkolev92) [inline proposed/2023/package-source-mapping-actions-PMUI.md]: > direct and transitive from selected source, direct and transitive from source different from the selected one and direct and  transitive  from a source not in the nuget.config
This is what I have from the chat from last Friday.

I don't think that's represented here.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**